### PR TITLE
Python: Pydantic settings config

### PR DESCRIPTION
**Pull request by eavanvalkenburg**
### Motivation and Context

The initial implementation of Pydantic Settings was not optimal yet, more optimized now.

### Description

Added a KernelBaseSettings class in kernel_pydantic
Rebased all settings off that
Added single create method to KernelBaseSettings:
- sets model_config.env_prefix to cls.end_prefix allowing us to set the env_prefix as a classvar on each subclass and it works
- if env_file_path is set, set that in model_config
- strip all None values out of inputted values, then instantiate

This also allows us to pass the settings that are supplied to the settings class, for overriding env settings.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/README.md">Py3.10 Test Coverage</a><details><summary>Python 3.10 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel/connectors/ai/google_palm/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_chat_completion.py">gp_chat_completion.py</a></td><td>70</td><td>17</td><td>76%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_chat_completion.py#L56-L57">56&ndash;57</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_chat_completion.py#L148-L153">148&ndash;153</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_chat_completion.py#L164-L165">164&ndash;165</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_chat_completion.py#L214">214</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_chat_completion.py#L217">217</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_chat_completion.py#L220-L221">220&ndash;221</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_chat_completion.py#L229">229</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_chat_completion.py#L233-L234">233&ndash;234</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_chat_completion.py#L242">242</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_text_completion.py">gp_text_completion.py</a></td><td>41</td><td>7</td><td>83%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_text_completion.py#L38-L39">38&ndash;39</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_text_completion.py#L70-L71">70&ndash;71</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_text_completion.py#L77-L78">77&ndash;78</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_text_completion.py#L107">107</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_text_embedding.py">gp_text_embedding.py</a></td><td>34</td><td>6</td><td>82%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_text_embedding.py#L36-L37">36&ndash;37</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_text_embedding.py#L63-L64">63&ndash;64</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/services/gp_text_embedding.py#L73-L74">73&ndash;74</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/google_palm/settings</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/settings/google_palm_settings.py">google_palm_settings.py</a></td><td>20</td><td>1</td><td>95%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/google_palm/settings/google_palm_settings.py#L43">43</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/open_ai/prompt_execution_settings</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/prompt_execution_settings/azure_chat_prompt_execution_settings.py">azure_chat_prompt_execution_settings.py</a></td><td>62</td><td>1</td><td>98%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/prompt_execution_settings/azure_chat_prompt_execution_settings.py#L95">95</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/open_ai/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py">azure_chat_completion.py</a></td><td>94</td><td>34</td><td>64%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py#L149">149</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py#L155-L156">155&ndash;156</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py#L165-L166">165&ndash;166</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py#L171-L188">171&ndash;188</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py#L192-L199">192&ndash;199</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py#L211-L222">211&ndash;222</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py">open_ai_chat_completion.py</a></td><td>24</td><td>2</td><td>92%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py#L55-L56">55&ndash;56</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_completion.py">open_ai_text_completion.py</a></td><td>25</td><td>3</td><td>88%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_completion.py#L56-L57">56&ndash;57</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_completion.py#L86">86</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py">open_ai_text_embedding.py</a></td><td>24</td><td>3</td><td>88%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py#L57-L58">57&ndash;58</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py#L87">87</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/open_ai/settings</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/settings/azure_open_ai_settings.py">azure_open_ai_settings.py</a></td><td>24</td><td>1</td><td>96%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/settings/azure_open_ai_settings.py#L76">76</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/settings/open_ai_settings.py">open_ai_settings.py</a></td><td>21</td><td>1</td><td>95%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/ai/open_ai/settings/open_ai_settings.py#L46">46</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/search_engine</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/search_engine/bing_connector.py">bing_connector.py</a></td><td>40</td><td>40</td><td>0%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/search_engine/bing_connector.py#L3-L84">3&ndash;84</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/search_engine/bing_connector_settings.py">bing_connector_settings.py</a></td><td>17</td><td>17</td><td>0%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/connectors/search_engine/bing_connector_settings.py#L3-L36">3&ndash;36</a></td></tr><tr><td colspan="5"><b>semantic_kernel/core_plugins/sessions_python_tool</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py">sessions_python_plugin.py</a></td><td>103</td><td>14</td><td>86%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py#L57-L59">57&ndash;59</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py#L79">79</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py#L82">82</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py#L88-L94">88&ndash;94</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py#L134">134</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py#L180">180</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_plugin.py#L184">184</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_settings.py">sessions_python_settings.py</a></td><td>31</td><td>1</td><td>97%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/a9ace6dfb8e8c3f3319f6214031167c4f59f14c0/python/semantic_kernel/core_plugins/sessions_python_tool/sessions_python_settings.py#L58">58</a></td></tr><tr><td><b>TOTAL</b></td><td><b>6078</b></td><td><b>870</b></td><td><b>86%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.10 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1374 | 1 :zzz: | 0 :x: | 0 :fire: | 16.194s :stopwatch: |

File: python/semantic_kernel/kernel_pydantic.py [hunk 1]
```diff
@@ -13,3 +14,40 @@ class KernelBaseModel(BaseModel):
...
+        case_sensitive=False,
+    )
+
+    @classmethod
+    def create(cls: Type["T"], **data: Any) -> "T":
+        """Update the model_config with the prefix."""
+        cls.model_config["env_prefix"] = cls.env_prefix
+        if data.get("env_file_path"):
+            cls.model_config["env_file"] = data["env_file_path"]
+            cls.model_config["env_file_encoding"] = "utf-8"
```
Comment (moonbox3) [inline python/semantic_kernel/kernel_pydantic.py:51]: For other locales that won't depend on utf-8, should we provide a way for them to set the `env_file_encoding`? For example Windows computers in China may use GBK.

Comment (eavanvalkenburg) [reply python/semantic_kernel/kernel_pydantic.py]: good point, added

**Review by moonbox3 — APPROVED** [review]

**Review by moonbox3 — APPROVED** [review]

File: python/semantic_kernel/connectors/ai/google_palm/services/gp_text_completion.py [hunk 2]
```diff
@@ -38,13 +38,22 @@ def __init__(
...
+        try:
+            google_palm_settings = GooglePalmSettings.create(
+                api_key=api_key,
+                text_model_id=ai_model_id,
+                env_file_path=env_file_path,
+                env_file_encoding=env_file_encoding,
+            )
+        except ValidationError as ex:
+            raise ServiceInitializationError("Failed to create Google Palm settings.", ex) from ex
+        if not google_palm_settings.text_model_id:
```
[the remark this answers is not in the document]
Comment (eavanvalkenburg) [reply python/semantic_kernel/connectors/ai/google_palm/services/gp_text_completion.py:54]: I thought about that, but that will require you to always set all three, while you might not need all three, so did this pattern, if there are required fields, catch the ValidationError, if there are semi-required fields, throw Service/MemoryConnector errors in code.

**Review by moonbox3 — APPROVED** [review]

**Review by moonbox3 — APPROVED** [review]

**Review by markwallace-microsoft — APPROVED** [review]

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**